File: split/makeleftright.pas
function makeleftright(q: halfword; style: smallnumber; maxd, maxh: scaled): smallnumber;
var
  delta, delta1, delta2: scaled;
begin
  if style < 4 then
    cursize := 0
  else
    cursize := 16 * ((style - 2) div 2);
  delta2 := maxd + fontinfo[22 + parambase[eqtb[3937 + cursize].hh.rh]].int;
  delta1 := maxh + maxd - delta2;
  if delta2 > delta1 then
    delta1 := delta2;
  delta := (delta1 div 500) * eqtb[5281].int;
  delta2 := delta1 + delta1 - eqtb[5840].int;
  if delta < delta2 then
    delta := delta2;
  mem[q + 1].int := vardelimiter(q + 1, cursize, delta);
  makeleftright := mem[q].hh.b0 - (10);
end;

File: cpp/src/makeleftright.cpp
#include "makeleftright.h"
#include "vardelimiter.h"
#include "police.h"

static int delimiter_factor(void) { return int_par(delimiter_factor_code); }
static int delimiter_shortfall(void) { return dimen_par(delimiter_shortfall_code); }

smallnumber makeleftright(halfword q, smallnumber style, scaled maxd, scaled maxh)
{
	if (style < 4)
		cursize = 0;
	else
		cursize = 16*((style-2)/2);
	scaled delta2 = maxd+axis_height(cursize);
	scaled delta1 = maxh+maxd-delta2;
	if (delta2 > delta1)
		delta1 = delta2;
	scaled delta = (delta1/500)*delimiter_factor();
	delta2 = 2*delta1-delimiter_shortfall();
	if (delta < delta2)
		delta = delta2;
	LeftRightNoad *Q;
	Q->num = q;
	new_hlist(q) = vardelimiter(Q->delimiter, cursize, delta)->num;
	return type(q)-(left_noad-open_noad);
}
